### PR TITLE
chore: jbang interactive release script

### PR DESCRIPTION
**Pull request by sunix**
## Description
First version of interactive script to release JKube

Usage: 
```
jbang ./scripts/jrelease.java
```

Fixes https://github.com/eclipse/jkube/issues/2202

To be improved in later PRs:
- checks before execution (if running jkube git root folder)
- non interactive mode
- better guessing for remote fork name
- automate some git actions with jgit
- error handling: should detect and fallback to terminal if error with a command

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [x] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: scripts/jrelease.java [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+        System.in.read();
+
+        // 17. Publish tweet in jkubeio profile
+        printStep(17, "Publish tweet in jkubeio profile");
+        // print the message to tweet:
+        // 🔥 @ECDTools #JKube new release 🚀
+        // 📦 v1.14.0
+        // ✨ Helidon support
+        // ✨ several improvements for helm
+        // ✨ CustomResource fragment improvements
```
Comment (rohanKanojia) [inline scripts/jrelease.java:188]: Do we want to keep these comments (these seem specific to previos release)?

File: scripts/jrelease.java [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+        // https://github.com/jkubeio/jkube-integration-tests/actions/workflows/smoke-tests.yml
+        printStep(20,
+                "Run the Gradle verification test: https://github.com/jkubeio/jkube-integration-tests/actions/workflows/smoke-tests.yml");
+        print("Press enter to continue");
+        System.in.read();
+
+    }
+
+    private static String retrieveCurrentVersionFromPomFile() {
+        String currentVersion = "1.14-SNAPSHOT";
```
Comment (rohanKanojia) [inline scripts/jrelease.java:239]: Why is it initialized to this value ?

**Comment by rohanKanojia** [comment]
I see read() calls here. Would this script require more than steps of input?

File: scripts/jrelease.java (hunk 1 above)
Comment (sunix) [reply scripts/jrelease.java:188]: no you are right, I used them as a template.
Actually we could remove almost all the comments WDYT ?

File: scripts/jrelease.java (hunk 2 above)
Comment (sunix) [reply scripts/jrelease.java:239]: you are right, maybe it should just fail if it cannot read

File: scripts/jrelease.java [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+///usr/bin/env jbang "$0" "$@" ; exit $?
+// //DEPS <dependency1> <dependency2>
+//DEPS org.jline:jline:3.16.0
+//DEPS de.codeshelf.consoleui:consoleui:0.0.13
+//DEPS org.fusesource.jansi:jansi:2.3.4
+//DEPS info.picocli:picocli:4.6.1
+//DEPS org.apache.maven:maven-model:3.8.1
```
Comment (manusa) [inline scripts/jrelease.java:20]: These might need an eclipse-dash check.

- Run the `eclipse-dash-tool.sh script.
- Then add these dependencies manually to the target/dependencies.txt list (following the same pattern.
- Then execute the last part of the script manually: 
  ```shell
  java -jar target/eclipse-dash.jar target/dependencies.txt -summary  target/dependencies-resolved.csv
  ```

**Review by manusa — CHANGES_REQUESTED** [review]
Please provide updates on the dependency check.

We might need to trigger clearlydefined harvesting for these dependencies manually

File: scripts/jrelease.java (hunk 3 above)
Comment (sunix) [reply scripts/jrelease.java:20]:
```
$ (jrelease) java -jar target/eclipse-dash.jar target/dependencies.txt -summary  target/dependencies-resolved.csv
[main] INFO Querying Eclipse Foundation for license data for 264 items.
[main] INFO Found 159 items.
[main] INFO Querying ClearlyDefined for license data for 105 items.
[main] INFO Found 105 items.
[main] INFO License information could not be automatically verified for the following content:
[main] INFO 
[main] INFO maven/mavencentral/de.codeshelf.consoleui/consoleui/0.0.13
[main] INFO 
[main] INFO This content is either not correctly mapped by the system, or requires review.
```

Comment (manusa) [reply scripts/jrelease.java:20]: I just harvested it, try to repeat and see if the java -jar command and see if it's OK now.

https://clearlydefined.io/definitions/maven/mavencentral/de.codeshelf.consoleui/consoleui/0.0.13

It has a score of 60, I can't remember if this was enough.

**Comment by sunix** [comment]
```
maven/mavencentral/org.jline/jline/3.16.0, BSD-3-Clause AND Apache-2.0 AND LicenseRef-Public-Domain, approved, CQ22622
maven/mavencentral/de.codeshelf.consoleui/consoleui/0.0.13, Apache-2.0, approved, clearlydefined
maven/mavencentral/org.fusesource.jansi/jansi/2.3.4, Apache-2.0, approved, clearlydefined
maven/mavencentral/info.picocli/picocli/4.6.1, Apache-2.0, approved, clearlydefined
maven/mavencentral/org.apache.maven/maven-model/3.8.1, Apache-2.0, approved, #2688

```

**Review by manusa — APPROVED** [review]

**Pull request merged**